### PR TITLE
Add Python 3 ast module; update Python 2.7 ast module; fixup typed_ast

### PR DESCRIPTION
**Pull request by ddfisher**

### CODE REVIEW

**Comment by gvanrossum** [comment]
@matthiaskramm What do you think?

@ddfisher Maybe separate the typed_ast changes from the stdlib changes?

**Comment by matthiaskramm** [comment]
Hm, so `2.7/ast.pyi` and `3/ast.pyi` differ only in what they import from `_ast.pyi`.

Switch to `from _ast import *` and merge into `2and3`?

File: stdlib/3/_ast.pyi [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+from typing import Any, Optional, Union
+
+PyCF_ONLY_AST = ... # type: int
+
+identifier = str
+
+class AST:
+    _attributes = ... # type: typing.Tuple[str, ...]
+    _fields = ... # type: typing.Tuple[str, ...]
+    def __init__(self, *args, **kwargs) -> None: ...
```
Comment (tharvik) [inline stdlib/3/_ast.pyi]: I guess you can add that (from https://docs.python.org/3/library/ast.html#ast.AST.col_offset), as `NameConstant` can pass `Any` as value. Also in stdlib/2/_ast.pyi.
```python
    def __init__(self, *args: Any, **kwargs: Any) -> None: ...
```

**Comment by gvanrossum** [comment]
Ping?

**Comment by gvanrossum** [comment]
Merged as-is in the hope of getting this into the mypy 0.4 release; please see if there's more to do.

**Pull request merged**